Implement a new logging system to allow OAA games to report to an external server

IBurn36360 opened the pull request:
This PR aims to address Issue #1443 

This is being opened up to allow for discussion on the manner in which this code is being developed for the OAA game side of things in case anything needs to change.

- Initial code shows:
    - inclusion placement (Very early in game init to allow the real game start to be captured as an event (This should allow games that hard crash or have full bails to be identified)).
    - One example call (Would capture and dispatch the game start event)
    - Full module for the internal to external reporting

chrisinajar commented:
we have print hooks in vconsole.lua, those might be useful

chrisinajar commented:
You can find example usage of the HTTP request object in the stats collection library in our codebase

IBurn36360 commented:
This is at a stage where a Loggly account needs to be set up and bound (Takes a matter of minutes to do).  I created one quickly just to test, and can likely delegate access out if no one wants to setup an account.

chrisinajar commented:
just make the account tokens configurable. if they're private then make it a dedicated config file, otherwise just put them into `settings.lua`

IBurn36360 commented:
The instance has an endpoint configuration internally at this time (I can move the actual references to come from the `settings.lua` file without an issue).

https://github.com/OpenAngelArena/oaa/pull/1486/commits/136d4be112b3bc68e4e64d2f721f33aee342e3d6#diff-027e60dec048b1b02390d99f12eba77cR29
https://github.com/OpenAngelArena/oaa/pull/1486/commits/136d4be112b3bc68e4e64d2f721f33aee342e3d6#diff-027e60dec048b1b02390d99f12eba77cR33

If that is all, then we might be good for a run on the test client.  I have, since the screencap shown in #general, added a players array with status and account name to help with debugging state at that point in time.

Since there shouldn't, ideally, be any errors in the normal release code, we might need to put in a testing error message just to see it working outside of the tools client.  I just added a test error call to the duel start function to test when in tools mode.

chrisinajar commented:
yeah it'd be great if it was boiled down to a `LOGGLY_ACCOUNT_ID` value in `settings.lua`

IBurn36360 commented:
Done and rebased over master so the merge conflict is handled

IBurn36360 commented:
Rebased to resolve version merge conflict

chrisinajar commented:
Yay! Super excited for this.

IBurn36360 commented:
Just before you go to merge this, I'll resolve the merge conflict (Is a side effect of the version constant being updated, so happens every version update).

IBurn36360 commented:
Since you could be merging this over the weekend, it is rebased over master to resolve the merge conflict (Also adds in a bail out case for if the Loggly account UID is not provided, allowing it to be merged before complete configuration without throwing errors when something goes to be sent).

chrisinajar commented:
Great work, thanks for adding the error checking code so I can merge this. Makes things easier :+1:
